src/nunit.analyzers/Extensions/INamespaceOrTypeSymbolExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace NUnit.Analyzers.Extensions
+{
+    internal static class INamespaceOrTypeSymbolExtensions
+    {
+        /// <summary>
+        /// Checks the full name of a type or namespace without incurring the expense of building a full name string.
+        /// </summary>
+        public static bool HasFullName(this INamespaceOrTypeSymbol symbol, params string[] segments)
+        {
+            if (segments.Length < 1)
+                throw new ArgumentException("At least one name segment must be specified.", nameof(segments));
+
+            if (segments.Any(string.IsNullOrEmpty))
```

Comment:
This check does allocate, right? (because of enumerator)

Reply:
Yes, and the `params` in the method signature allocates. I was actively avoiding string building more than allocations in general, and even more specifically thinking of this method as a helper to automate doing the pattern I see in Roslyn:

```cs
var isTypeConverterAttribute = type is
{
    Name: "TypeConverterAttribute",
    ContainingSymbol: INamespaceSymbol
    {
        Name: "ComponentModel",
        ContainingNamespace:
        {
            Name: "System",
            ContainingNamespace: { IsGlobalNamespace = true },
        }
    },
};
```